ethcore/src/engines/validator_set/contract.rs

```diff
@@ -58,6 +59,7 @@ impl ValidatorContract {
 
 		match client.as_full_client() {
 			Some(c) => {
+				c.call_contract_as_author(BlockId::Latest, self.contract_address.clone(), data.clone())?;
```

Comment:
Two things here:
1. Maybe I'm missing something, but why do we need to add this?
2. I don't think the `.clone()` on the contract address is necessary

Reply:
1. It's used exactly for checking whether the tx will be reverted or not and if yes it will fail with an error.
2. Done :)